### PR TITLE
Introduce AggregateMetrics dataclass

### PR DESCRIPTION
**Pull request by dvadym**
Before this CL, the output of UtilityAnalysis is list of metrics of the format:

**Private partitions:** [PartitionSelectionMetrics, AggregateErrorMetrics, PartitionSelectionMetrics `...]`
Where each consecutive pair `PartitionSelectionMetrics` and `AggregateErrorMetrics` correspond to one Utility Analysis configuration (UtilityAnalysis can run for multiple parameters simultaneously, e.g. for differemt max_partition_contributed).

**Public partitions**: `[AggregateErrorMetrics, AggregateErrorMetrics ...]`

**Notes:**
1. `PartitionSelectionMetrics` contains partition selection metrics (e.g. the expected number of partitions)
2. `AggregateErrorMetrics` contains error per partitions (e.g. average error per partition)

Having the output in different format for private and public partitions is very inconvenient and requires calling code to do this processing. This PR introduced class `AggregateMetrics` which contains those both `PartitionSelectionMetrics` and `AggregateErrorMetrics`.

### CODE REVIEW

File: utility_analysis_new/parameter_tuning.py [hunk 1]
```diff
@@ -151,27 +151,19 @@ def _convert_utility_analysis_to_tune_result(
     MultiParameterConfiguration, use_public_partitions: bool,
         contribution_histograms: histograms.ContributionHistograms
 ) -> TuneResult:
-
-    # Get only error metrics, ignore partition selection for now.
-    # TODO: Make the output of the utility analysis 1 dataclass per 1 run.
-    if use_public_partitions:
```
Comment (dvadym) [inline utility_analysis_new/parameter_tuning.py:157]: With the AggregateErrors, no need to do this.

**Review by Kulankhina — APPROVED** [review]

File: utility_analysis_new/utility_analysis.py [hunk 2]
```diff
@@ -89,12 +90,34 @@ def perform_utility_analysis(col,
...
+    aggregates = backend.values(aggregates, "Drop key")
+    # aggregates: (accumulator)
+
+    aggregates = backend.map(aggregates,
+                             aggregate_error_combiners.compute_metrics,
+                             "Compute aggregate metrics")
+
+    # accumulators : (aggregate_metrics)
+
+    def pack_metrics(aggregate_metrics):
```
[the remark this answers is not in the document]
Comment (dvadym) [reply utility_analysis_new/utility_analysis.py]: Type annotations should be added, whenever it's easy to do :)

Thanks, done.

File: utility_analysis_new/utility_analysis.py [hunk 3]
```diff
@@ -89,12 +90,34 @@ def perform_utility_analysis(col,
...
+            # AggregateErrorMetrics, PartitionSelectionMetrics ...]
+            # Each consecutive pair PartitionSelectionMetrics and
+            # AggregateErrorMetrics correspond to one Utility Analysis
+            # configuration.
+            return [
+                metrics.AggregateMetrics(aggregate_metrics[i + 1],
+                                         aggregate_metrics[i])
+                for i in range(0, len(aggregate_metrics), 2)
+            ]
+        return [metrics.AggregateMetrics(m) for m in aggregate_metrics]
```
[the remark this answers is not in the document]
Comment (dvadym) [reply utility_analysis_new/utility_analysis.py]: done

**Review by dvadym — COMMENTED** [review]
Thanks for review!

**Pull request merged**